romancal/tessel_match/tessel_match.py

```diff
@@ -0,0 +1,291 @@
+                " or pixel_shape attributes be set to the image shape or that the"
+                "image_shape argument be set"
+            )
+        if image_shape is not None:
+            pass
+        else:
+            if hasattr(iwcs, "bounding_box") and iwcs.bounding_box is not None:
+                # Presumes that the bounding_box matches image array boundaries
+                bbintervals = iwcs.bounding_box.intervals
+                # This compensates for the half pixel adjustment in the general code.
+                image_shape = (bbintervals[0].upper + 0.5, bbintervals[1].upper + 0.5)
+            elif hasattr(iwcs, "pixel_shape") and iwcs.pixel_shape is not None:
+                image_shape = iwcs.pixel_shape
+        # Compute the image corners ra, dec from the wcs
+        (cxm, cxp), (cym, cyp) = (
+            (-0.5, image_shape[1] - 0.5),
+            (-0.5, image_shape[0] - 0.5),
+        )
+        image_corners = (iwcs(cxp, cyp), iwcs(cxm, cyp), iwcs(cxm, cym), iwcs(cxp, cym))
+    ttab = TESSEL_TABLE
+    ra = ttab[:]["ra_center"]
+    dec = ttab[:]["dec_center"]
+    # # Convert all celestial coordinates to cartesion coordinates.
+    vec_centers = np.array(sgv.lonlat_to_vector(ra, dec)).transpose()
+    # # Organize corners into two ra, dec lists
+    # imra = np.array([point[0] for point in image_corners])
+    # imdec = np.array([point[1] for point in image_corners])
+    vec_im_corners = image_coords_to_vec(image_corners)
+    # Approximate center of image by averaging corner vectors
+    im_center = normalize_vector(vec_im_corners.mean(axis=1))
+    # Compute difference vector between image centerand tessel centers
+    diff = vec_centers - im_center
+    dist = np.sqrt((diff**2).sum(axis=1)) * 180 / np.pi
+    match = np.where(dist < 0.5)
+    ncandidates = len(match[0])
+    # Now see which of these that are close actually overlap the supplied image.
+    # (Is it necessary to check that the corners are in a sensible order?)
+    mra1 = ttab[match]["ra_corn1"]
+    mra2 = ttab[match]["ra_corn2"]
+    mra3 = ttab[match]["ra_corn3"]
+    mra4 = ttab[match]["ra_corn4"]
+    mdec1 = ttab[match]["dec_corn1"]
+    mdec2 = ttab[match]["dec_corn2"]
+    mdec3 = ttab[match]["dec_corn3"]
+    mdec4 = ttab[match]["dec_corn4"]
+    mcenters = vec_centers[match]
+    mra = np.vstack([mra1, mra2, mra3, mra4, mra1])
+    mdec = np.vstack([mdec1, mdec2, mdec3, mdec4, mdec1])
+    points = np.array(sgv.lonlat_to_vector(mra, mdec))
+    # Create polygon for supplied image_corners
+    pvec_im_corners = np.zeros((5, 3))
+    pvec_im_corners[:4] = vec_im_corners.transpose()
+    pvec_im_corners[4] = pvec_im_corners[0]
+    impoly = sgp.SingleSphericalPolygon(pvec_im_corners, im_center)
+    realmatch = []
+    for i in range(ncandidates):
+        # print(i)
+        cellpoints = points[:, :, i].transpose()
+        cellcenter = mcenters[i]
+        cellpoly = sgp.SingleSphericalPolygon(cellpoints, cellcenter)
```

Comment:
Mostly as a note to myself: for tangent plane projections, I think the edges of the skycells are "small circles" rather than "great circles" and so they are not exactly described by a spherical polygon.  The image itself has some other non-trivial border shape.  This will be a good approximation but not exact.

Reply:
I suspected as much, but referring to the previous discussion, it would be simpler just to pad the sides by the worst case.